macros/edr/materializations/tests/test.sql

```diff
@@ -0,0 +1,242 @@
+{% macro query_test_result_rows(sample_limit=none) %}
+  {%- set query -%}
+      select * from ({{ sql }}) {%- if sample_limit -%} limit {{ sample_limit }} {%- endif -%}
+  {%- endset -%}
+  {% do return(elementary.agate_to_dicts(dbt.run_query(query))) %}
+{% endmacro %}
+
+{% macro cache_elementary_test_results_rows(elementary_test_results_rows) %}
+  {% do elementary.get_cache("elementary_test_results").update({model.unique_id: elementary_test_results_rows}) %}
+{% endmacro %}
+
+
+{% macro handle_anomaly_test(flattened_test) %}
+  {% set metrics_tables_cache = elementary.get_cache("tables").get("metrics") %}
+  {% set metrics_table = elementary.get_elementary_test_table(flattened_test.name, 'metrics') %}
+  {% if metrics_table %}
+    {% do metrics_tables_cache.append(metrics_table) %}
+  {% endif %}
+
+  {% set anomaly_scores_rows = elementary.query_test_result_rows() %}
+  {% if anomaly_scores_rows and flattened_test.short_name == "dimension_anomalies" %}
```

Comment:
I think it will be good to add a comment (or even split into smaller macros) to explain why we handle dimension like that, and why we handle other anomalies different.

This is the most important part as you said, and we should make it as easy as possible for someone to understand the logic behind it 🙂